Source/WebKit/UIProcess/API/haiku/WebView.cpp

```diff
@@ -127,8 +132,9 @@ void BWebView::didCommitNavigation(WKPageRef page, WKNavigationRef navigation, W
 }
 void BWebView::didReceiveServerRedirectForProvisionalNavigation(WKPageRef page, WKNavigationRef navigation, WKTypeRef userData, const void* clientInfo)
 {
-	//this wont work for now as we need to make network process work
-	//fprintf(stderr,"\n %s %s\n",__PRETTY_FUNCTION__,((BWebView*)clientInfo)->getCurrentURL());
+	BMessage message(URL_CHANGE);
+	message.AddString("url",BString(((BWebView*)clientInfo)->getCurrentURL()));
+	be_app->PostMessage(&message);
```

Comment:
Why does this goes to be_app? It should go to the BWindow the view is part of (you can get it using BLooper* window = Looper();)